### PR TITLE
allow numeric components in snake case

### PR DESCRIPTION
**Pull request by acunniffe**
Fixes https://github.com/snyk/sweater-comb/issues/238

Updated regex should allow names like: `something_in_first_30_days`, but will not allow `30_day_setup`. 

@cmars Is this correct behavior or do you want it to allow digits in all components? 

I defaulted to this approach first because most programming languages do not like variable names that start with digits. It can be annoying for consumers who may discover they need to do some additional mapping while marshalling/unmarshalling to get compilers to work.

### CODE REVIEW

File: src/rulesets/properties.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export const rules = {
   propertyKey: ({ bodyProperties }: SnykApiCheckDsl) => {
     bodyProperties.added.must("have snake case keys", ({ key }, context) => {
       // TODO: did not find a doc link for this
-      const snakeCase = /^[a-z]+(?:_[a-z]+)*$/g;
+      const snakeCase = /^[a-z]+(?:_[a-z\d]+)*$/g;
```
Comment (acunniffe) [inline src/rulesets/properties.ts:59]: small change

Comment (acunniffe) [reply src/rulesets/properties.ts:66]: Matchers within square brackets are called character classes. A character range is valid inside of a character class. So are the shorthands like `\d`. They're meant to be mixed. source https://www.regular-expressions.info/charclass.html

**Comment by acunniffe** [comment]
@cmars added tests, missed the notif.

**Review by cmars — APPROVED** [review]
LGTM & thanks for this! Looks like CI is failing on lint, this just needs a `yarn run format`.

**Pull request merged**